Packs/TAXIIServer/Integrations/TAXII2Server/TAXII2Server_test.py

```diff
@@ -0,0 +1,466 @@
+        Then
+            Validate that the error and status code right
+    """
+    mocker.patch('TAXII2Server.SERVER', taxii2_server_v21)
+    mocker.patch.object(demisto, 'error')
+    mocker.patch.object(demisto, 'updateModuleHealth')
+    with APP.test_client() as test_client:
+        response = test_client.get('/threatintel/collections/not_exsisting_collection_id/', headers=HEADERS)
+        assert response.status_code == 404
+        assert response.json.get('title') == 'Unknown Collection'
+
+
+def test_taxii20_manifest(mocker, taxii2_server_v20):
+    """
+        Given
+            TAXII Server v2.0, collection_id, range
+        When
+            Calling manifest api request for given collection
+        Then
+            Validate that right manifest returned.
+    """
+    iocs = util_load_json('test_files/ip_iocs.json')
+    manifest = util_load_json('test_files/manifest20.json')
+    headers = copy.deepcopy(HEADERS)
+    headers['Range'] = 'items 0-4'
+    mocker.patch('TAXII2Server.SERVER', taxii2_server_v20)
+    mocker.patch.object(demisto, 'searchIndicators', return_value=iocs)
+    mocker.patch.object(demisto, 'params', return_value={'res_size': '100'})
+    with APP.test_client() as test_client:
+        response = test_client.get('/threatintel/collections/4c649e16-2bb7-50f5-8826-2a2d0a0b9631/manifest/',
+                                   headers=headers)
+        assert response.status_code == 200
+        assert response.content_type == 'application/vnd.oasis.taxii+json; version=2.0'
+        assert response.json == manifest
+
+
+def test_taxii21_manifest(mocker, taxii2_server_v21):
+    """
+        Given
+            TAXII Server v2.1, collection_id
+        When
+            Calling manifest api request for given collection
+        Then
+            Validate that right manifest returned.
+    """
+    iocs = util_load_json('test_files/ip_iocs.json')
+    manifest = util_load_json('test_files/manifest21.json')
+    mocker.patch.object(demisto, 'params', return_value={'res_size': '100'})
+    mocker.patch('TAXII2Server.SERVER', taxii2_server_v21)
+    mocker.patch.object(demisto, 'searchIndicators', return_value=iocs)
+    with APP.test_client() as test_client:
+        response = test_client.get('/threatintel/collections/4c649e16-2bb7-50f5-8826-2a2d0a0b9631/manifest/?limit=4',
+                                   headers=HEADERS)
+        assert response.status_code == 200
+        assert response.content_type == 'application/taxii+json;version=2.1'
+        assert response.json == manifest
+
+
+def test_taxii20_objects(mocker, taxii2_server_v20):
+    """
+        Given
+            TAXII Server v2.0, collection_id, content-range
+        When
+            Calling get objects api request for given collection
+        Then
+            Validate that right manifest returned.
+    """
+    iocs = util_load_json('test_files/ip_iocs.json')
+    objects = util_load_json('test_files/objects20.json')
```

Comment:
there are 5 objects inside ip_iocs.json

Reply:
I meant why there are four, but I understood they are the extension objects. but why do we have them if has_exstension is false? (nothing is configured in `fields_filter` param)

Reply:
has_exstension is true. 
It is because: 
![image](https://user-images.githubusercontent.com/72339940/147951948-f6b0e335-a11c-45d6-9e99-43156574ac02.png)
and fields_to_present = set() and not {name, type}